File: yunva17/프로그래머스/Lv2/구명보트.js
function solution(people, limit) {
  let answer = 0;
  people = people.sort((a, b) => a - b);

  let start = 0;
  let end = people.length - 1;

  while (start <= end) {
    if (people[start] + people[end] <= limit) {
      start++;
      end--;
    } else {
      end--;
    }
    answer++;
  }

  return answer;
}

File: pbkdpwls/그리디/구명보트.java
import java.util.*;

class Solution {
    public int solution(int[] people, int limit) {
        int answer = 0;
        int max = people.length-1; // max 값 인덱스
        int min=0; // min 값 인덱스

        Arrays.sort(people);

        while(max>=0 && max>=min){
            if(people[min]+people[max]<=limit) {
                answer++;
                min++;

                if(max==1){
                    break;
                }
            }
            else {
                answer++;
            }
            max--;
        }

        return answer;
    }
}